lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php

```diff
@@ -53,8 +58,71 @@ public function __construct(Reflector $reflector, BuilderFactory $factory, Updat
         $this->factory = $factory;
     }
 
-    public function extractMethod(SourceCode $source, int $offsetStart, int $offsetEnd, string $name): SourceCode
+    public function canExtractMethod(SourceCode $source, int $offsetStart, int $offsetEnd): bool
     {
+        if ($offsetEnd == $offsetStart) {
+            return false;
+        }
+        $node = $this->parser->parseSourceFile($source->__toString());
+        $endNode = $node->getDescendantNodeAtPosition($offsetEnd);
+        $startNode = $node->getDescendantNodeAtPosition($offsetStart);
+
+        if (
+            $endNode instanceof CompoundStatementNode &&
+            $endNode->openBrace->getEndPosition() < $offsetEnd &&
+            $endNode->closeBrace->getStartPosition() >= $offsetEnd &&
+            count($endNode->statements) > 0
+        ) {
+            $stmt = end($endNode->statements);
+            assert($stmt instanceof Node);
+            while ($stmt && $stmt->getEndPosition() > $offsetEnd) {
+                $stmt = prev($endNode->statements);
+            }
+            
+            $endNode = $stmt ?? $endNode;
+            assert($endNode instanceof Node);
+        }
+
+        while ($endNode->parent && !($endNode->parent instanceof CompoundStatementNode)) {
+            $endNode = $endNode->parent;
+        }
+
+        if (
+            $startNode instanceof CompoundStatementNode &&
+            $startNode->openBrace->getEndPosition() <= $offsetStart &&
+            $startNode->closeBrace->getStartPosition() > $offsetStart &&
+            count($startNode->statements) > 0
+        ) {
+            $stmt = current($startNode->statements);
+            assert($stmt instanceof Node);
+            while ($stmt && $stmt->getStart() < $offsetStart) {
+                $stmt = next($startNode->statements);
+            }
+            
+            if ($stmt === false) {
+                dump($startNode->getStart(), $startNode->getEndPosition(), $offsetStart);
```

Comment:
diump

Reply:
shoudl this return false?

Reply:
What do you mean?

Reply:
what should this do, should it `return false`

Reply:
No, I was trying to see when does this case occur. The point of the validation is to ensure that the start and end offsets are in the same code block: 
```php
{
  ...
  <>
  ...
  <>
  ...
}
```
and there are multiple cases related to that depending on where the selection starts and ends. That case concerns the case "empty block", e.g.:
```php
if {
  <>
}
```
The the start statement is the block itself. It's a valid statement to be extracted, so it doesn't need to return `false`.